### PR TITLE
treewide: Address uncrustify suggestions

### PR DESCRIPTION
**Pull request by JKRhb**
### Contribution description

This PR makes a couple of minor changes to get rid of the uncrustify suggestions made in the CI.

The changes in `sys/include/ztimer.h` are the result of false positives and can probably be dropped. Maybe there is a way to adjust the config to get rid of warnings like these?

### Testing procedure

All static tests should pass without suggestions/warnings while the CI stays green as well.

### Issues/PRs references

The false postives for `sys/include/ztimer.h` have been mentioned in #15802 before.

### CODE REVIEW

File: sys/include/ztimer.h [hunk 1]
```diff
@@ -305,7 +305,7 @@ typedef struct {
      * @param   clock       ztimer clock to set the new target
      * @param   val         Relative target (e.g. fire at value `now() + val`)
      */
-    void (*set)(ztimer_clock_t *clock, uint32_t val);
+    void (*set)(ztimer_clock_t * clock, uint32_t val);
```
Comment (benpicco) [inline sys/include/ztimer.h:308]: shouldn't the proper style be
```suggestion
    void (*set)(ztimer_clock_t* clock, uint32_t val);
```

File: sys/include/ztimer.h [hunk 2]
```diff
@@ -317,7 +317,7 @@ typedef struct {
      * @brief   Cancel any set target
      * @param   clock       ztimer clock to cancel a pending alarm, if any
      */
-    void (*cancel)(ztimer_clock_t *clock);
+    void (*cancel)(ztimer_clock_t * clock);
```
Comment (benpicco) [inline sys/include/ztimer.h:320]:
```suggestion
    void (*cancel)(ztimer_clock_t* clock);
```

File: sys/include/ztimer.h (hunk 1 above)
Comment (JKRhb) [reply sys/include/ztimer.h:308]: Hmm, I am not sure, everywhere else the star is put before the parameter name (even in the case of the third function in this struct, see line 314). What is very odd is the fact that the star placement seems to only fail in the case of functions with return type `void`. If you change the return type from `void` to `uint32_t`, for example, the uncrustification works as expected 🤔

Comment (benpicco) [reply sys/include/ztimer.h:308]: Uh yea of course - `ztimer_clock_t *clock` is correct, not `ztimer_clock_t* clock`.
Not sure why uncrustify insists on putting a space here.

Comment (JKRhb) [reply sys/include/ztimer.h:308]: I just did some investigation and noticed that the uncrustify package [used in the riotdocker image](https://github.com/RIOT-OS/riotdocker/runs/2773139817?check_suite_focus=true#step:5:803) (and installed under ubuntu using the default package sources) is version 0.66.0 which dates back to 2017 (current version is 0.73.0. Updating uncrustify to the latest version should probably fix this issue. I guess we can leave this as is for now as a reminder?

Comment (benpicco) [reply sys/include/ztimer.h:308]: https://github.com/RIOT-OS/riotdocker/pull/139 should help

Comment (JKRhb) [reply sys/include/ztimer.h:308]: Oh, very nice! :) @kaspar030: Can you estimate when that PR will be ready?

Comment (kaspar030) [reply sys/include/ztimer.h:308]: I guess someone needs to review it ;)

Comment (JKRhb) [reply sys/include/ztimer.h]: With the new uncrustify version the original version of this line does not get flagged anymore :) I therefore reverted the change.

File: cpu/riscv_common/irq_arch.c [hunk 3]
```diff
@@ -82,13 +82,13 @@ void riscv_irq_init(void)
 /**
  * @brief Global trap and interrupt handler
  */
-static void __attribute((used)) handle_trap(uint32_t mcause)
-{
+static void __attribute((used)) handle_trap(uint32_t mcause){
```
Comment (benpicco) [inline cpu/riscv_common/irq_arch.c:85]: Shouldn't that be 
```suggestion
__attribute((used))
static void handle_trap(uint32_t mcause)
{
```

File: cpu/riscv_common/irq_arch.c [hunk 4]
```diff
@@ -149,8 +151,7 @@ static void __attribute((used)) handle_trap(uint32_t mcause)
 /* Marking this as interrupt to ensure an mret at the end, provided by the
  * compiler. Aligned to 64-byte boundary as per RISC-V spec and required by some
  * of the supported platforms (gd32)*/
-static void __attribute((aligned(64))) __attribute__((interrupt)) trap_entry(void)
-{
+static void __attribute((aligned(64))) __attribute__((interrupt)) trap_entry(void){
```
Comment (benpicco) [inline cpu/riscv_common/irq_arch.c:154]: same here

Comment (JKRhb) [reply cpu/riscv_common/irq_arch.c:154]: Hmm, like so?

```suggestion
__attribute((aligned(64)))
static void __attribute__((interrupt)) trap_entry(void){
```

(I am not entirely sure about the difference between `__attribute__`s and `__attribute__`s yet.)

**Comment by fjmolinas** [comment]
ping @benpicco @JKRhb lets not let this one diverge

**Comment by JKRhb** [comment]
Rebased and removed some false positives that were caused by the old uncrustify version :)

**Review by benpicco — APPROVED** [review]

**Comment by JKRhb** [comment]
@benpicco Shall I squash?

**Comment by JKRhb** [comment]
@benpicco Done :) I took the freedom of adding the "ready for build" label, I hope that's okay.

File: cpu/riscv_common/periph/plic.c [hunk 5]
```diff
@@ -30,8 +30,12 @@
...
 
 /* Local macros to calculate register offsets */
-#define _REG32(p, i) 			(*(volatile uint32_t *) ((p) + (i)))
-#define PLIC_REG(offset) 		_REG32(PLIC_CTRL_ADDR, offset)
+#ifndef _REG32
+#define _REG32(p, i)            (*(volatile uint32_t *)((p) + (i)))
+#endif
+#ifndef PLIC_REG
+#define PLIC_REG(offset)        _REG32(PLIC_CTRL_ADDR, offset)
+#endif
```
Comment (JKRhb) [inline cpu/riscv_common/periph/plic.c:38]: I added these `#ifndef`s as both macros are also defined in vendor files and the reformatting led to redefinition errors as the whitespacing is not the same anymore. I hope this approach is feasible.

File: sys/ztimer/convert.c [hunk 6]
```diff
@@ -42,7 +42,7 @@ void ztimer_convert_init(ztimer_convert_t *ztimer_convert,
     ztimer_convert_t tmp = {
         .lower = lower,
         .lower_entry = {
-            .callback = (void (*)(void *))ztimer_handler,
+            .callback = (void (*)(void *)) ztimer_handler,
```
Comment (maribu) [inline sys/ztimer/convert.c:45]: Note: This suggestion from `uncrustify` is only due to `uncrustify` not being able to reliably tell the unary * operator and the pointer type * apart from the binary * operator.

Casts have to be placed like a unary operator directly in from of the variable. So this was actually correct before and should be left as it is.

File: sys/ztimer/convert_frac.c [hunk 7]
```diff
@@ -97,7 +97,7 @@ void ztimer_convert_frac_init(ztimer_convert_frac_t *self,
         .super.super = { .ops = &ztimer_convert_frac_ops, },
         .super.lower = lower,
         .super.lower_entry =
-        { .callback = (void (*)(void *))ztimer_handler, .arg = &self->super, },
+        { .callback = (void (*)(void *)) ztimer_handler, .arg = &self->super, },
```
Comment (maribu) [inline sys/ztimer/convert_frac.c:100]: Same

**Review by maribu — COMMENTED** [review]
See inline

File: sys/ztimer/convert.c (hunk 6 above)
Comment (JKRhb) [reply sys/ztimer/convert.c]: Thank you! I dropped the commit where these changes were introduced :)

**Pull request merged**